interview-questions.md

```diff
@@ -8,62 +8,60 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is the difference between a parameter and an argument?
 
-  Your answer:
+  Your answer: A param is a variable or set of variables in a function. An arugment is the function its self.
 
-  Researched answer:
+  Researched answer: Inside a function, a parameter is the placeholder or template. An arugment is the actual content of said placeholder.
 
 
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer:
-
-  Researched answer:
-
-
+  Your answer: They are the parameters you have specifiyed in the argument.Not sure on which are required vs optional.
 
+  Researched answer:.maps are used for iteration in an array. They go through the array usually effecting it some how and returning another array after the effected process. What is required is the value of the array, what is optional iss the index of the array.
+  
+  
 3. What is the difference between map and filter?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: Map iterates and returns all parts of the array. Filter only returns specific parts of the origional filter.
 
+  Researched answer: Map takes all objects in an array to apply an effect to. Filter takes al objects but creates a new / different array after.
 
 
 4. What is the difference between a function and a method?
 
-  Your answer:
+  Your answer: A function is the whole process of doing something. A method is a smaller part of that fuction doing a more specific process.
 
-  Researched answer:
+  Researched answer: A function is a piece of code that can be given data to process and return. A method operates on information that is contanied within a class and uses a object to process.
 
 
 
 5. What is object destructuring?
 
-  Your answer:
+  Your answer: Can't quite remember but I would guess it has to do with disassembling an object.
 
-  Researched answer:
+  Researched answer: It is a way to assign variables to an object. It lets you be able to assing individual values to something like an arrary or object. Breaking an object into individual variables.
 
 
 
 6. STRETCH: What is hoisting in JavaScript?
 
-  Your answer:
+  Your answer: No really sure. Don't recall hearing that term.
```

Comment:
Yeah we didn't talk about this one directly. Just an opportunity to keep learning.